src/test/java/io/confluent/connect/jdbc/source/JdbcSourceTaskUpdateTest.java

```diff
@@ -366,11 +475,38 @@ public void testTimestampRestoreOffset() throws Exception {
     PowerMock.verifyAll();
   }
 
+  @Test
+  public void testTimestampAndIncrementingRestoreNoVersionOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), 3L);
+    testTimestampAndIncrementingRestoreOffset(
+        Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap())
+    );
+  }
+
+  @Test
+  public void testTimestampAndIncrementingRestoreVersionOneOffset() throws Exception {
+    TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), 3L);
+    testTimestampAndIncrementingRestoreOffset(
+        Collections.singletonMap(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap())
+    );
+  }
+
   @Test
   public void testTimestampAndIncrementingRestoreOffset() throws Exception {
+    TimestampIncrementingOffset oldOffset = new TimestampIncrementingOffset(new Timestamp(10L), 2L);
     TimestampIncrementingOffset offset = new TimestampIncrementingOffset(new Timestamp(10L), 3L);
-    expectInitialize(Arrays.asList(SINGLE_TABLE_PARTITION),
-                     Collections.singletonMap(SINGLE_TABLE_PARTITION, offset.toMap()));
+    Map<Map<String, String>, Map<String, Object>> offsets = new HashMap<>();
+    offsets.put(SINGLE_TABLE_PARTITION_WITH_VERSION, offset.toMap());
+    offsets.put(SINGLE_TABLE_PARTITION, oldOffset.toMap());
+    testTimestampAndIncrementingRestoreOffset(offsets);
```

Comment:
How about a comment here to say that we want to always use the offset with the latest protocol found?